packages/core/src/node/process/process.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable } from "inversify";
+import { ILogger } from '../../common/logger';
+import * as child from 'child_process';
+import * as stream from 'stream';
+import { Emitter, Event } from '../../common/event';
+import { Disposable } from '../../common/disposable';
+
+export interface IProcessExitEvent {
+    code: number,
+    signal?: string
+}
+
+@injectable()
+export abstract class Process implements Disposable {
+
+    abstract readonly type: 'Raw' | 'Terminal';
+    abstract pid: number;
+    abstract output: stream.Readable;
+    protected abstract process: child.ChildProcess | undefined;
+    protected abstract terminal: any;
+    protected readonly exitEmitter = new Emitter<IProcessExitEvent>();
+    protected readonly errorEmitter = new Emitter<Error>();
+
+    constructor(protected readonly logger: ILogger) {
+        this.exitEmitter.event(this.handleOnExit.bind(this));
+        this.errorEmitter.event(this.handleOnError.bind(this));
+    }
+
+    abstract kill(signal?: string): void;
+
+    get onExit(): Event<IProcessExitEvent> {
+        return this.exitEmitter.event;
+
+    }
+
+    get onError(): Event<Error> {
+        return this.errorEmitter.event;
+
+    }
+
+    dispose() {
+        this.exitEmitter.dispose();
```

Comment:
Should not be process stopped?

Reply:
Yeah I think it should, I was managing that externally before but I think it would be less confusing to use dispose for that, good point